src/documentation/TableWithFilter/Form.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import { useForm, Controller } from 'react-hook-form';
+import FormGroup from '../../components/FormGroup';
+import TextInput from '../../components/TextInput';
+import NumberInput from '../../components/NumberInput';
+import Button from '../../components/Button';
+import ReactSelect from 'react-select';
+import ReactSelectWrapper from './ReactSelectWrapper';
+
+const options = [
+    { value: 'male', label: 'Male' },
+    { value: 'female', label: 'Female' },
+  ];
+const countryOptions = [
+    
+    { value: 'armenia', label: 'Armenia' },
+    { value: 'brazil', label: 'Brazil' },
+    { value: 'china', label: 'China' },
+    { value: 'france', label: 'France' },
+    { value: 'honduras', label: 'Honduras' },
+    { value: 'indonesia', label: 'Indonesia' },
+    { value: 'morocco', label: 'Morocco' },
+    { value: 'pakistan', label: 'Pakistan' },
+    { value: 'russia', label: 'Russia' },
+    { value: 'zimbabwe', label: 'Zimbabwe' },
+    
+]
+
+function Form({ onFormChange }) {
+    const { handleSubmit, register, errors, control } = useForm();
+
+    const handleFilter = (formdata) => {
+        onFormChange(formdata)
+    }
+
+
+    return (
+        <form onSubmit={handleSubmit(handleFilter)}>
+            <FormGroup className="wfp--form-long"
+                align="horizontal"
+                style={{ marginTop: '1rem' }}>
+                <div className="wfp--form-item" style={{ minWidth: '100px' }}>
+                <Controller
+                    as={<ReactSelectWrapper control={control} options={options} />}
+                    name="gender"
+                    labelText="Gender"
+                    control={control}
+                />
+                </div>
+                <div className="wfp--form-item" style={{ minWidth: '100px' }}>
+                <Controller
+                    as={<ReactSelectWrapper control={control} options={countryOptions} />}
+                    name="country"
+                    labelText="Country"
+                    control={control}
+                />
+                </div>
+                
+            </FormGroup>
+            <FormGroup className="wfp--form-long" align="horizontal" style={{ marginTop: '1rem' }}>
+                <TextInput
+                id="firstname"
```

Comment:
The prop `id` might no longer be required. In case there is no id defined it will take the name prop as id if I recall correctly.

Reply:
yes, it will take the name prop as id

Reply:
If id is not defined it will automatically take the name as id.